scipy/sparse/csgraph/tests/test_graph_laplacian.py

```diff
@@ -99,86 +100,255 @@ def _assert_allclose_sparse(a, b, **kwargs):
     assert_allclose(a, b, **kwargs)
 
 
-def _check_laplacian(A, desired_L, desired_d,
-                     normed, use_out_degree, copy, dtype, arr_type):
+def _check_laplacian_dtype_none(
+    A, desired_L, desired_d, normed, use_out_degree, copy, dtype, arr_type
+):
+    mat = arr_type(A, dtype=dtype)
+    L, d = csgraph.laplacian(
+        mat,
+        normed=normed,
+        return_diag=True,
+        use_out_degree=use_out_degree,
+        copy=copy,
+        dtype=None,
+    )
+    if normed and (
+        np.issubdtype(mat.dtype, np.signedinteger)
+        or np.issubdtype(mat.dtype, np.uint)
+    ):
+        assert L.dtype == np.float64
+        assert d.dtype == np.float64
+        _assert_allclose_sparse(L, desired_L, atol=1e-12)
+        _assert_allclose_sparse(d, desired_d, atol=1e-12)
+    else:
+        assert L.dtype == dtype
+        assert d.dtype == dtype
+        desired_L = np.asarray(desired_L).astype(dtype)
+        desired_d = np.asarray(desired_d).astype(dtype)
+        _assert_allclose_sparse(L, desired_L, atol=1e-12)
+        _assert_allclose_sparse(d, desired_d, atol=1e-12)
+
+    if not copy:
+        if not (
+            normed
+            and (
+                np.issubdtype(mat.dtype, np.signedinteger)
+                or np.issubdtype(mat.dtype, np.uint)
+            )
+        ):
```

Comment:
Maybe this is a sign that we should pull out this expression into a little helper function, because the black formatting is really awkward here (and elsewhere in this file).

Reply:
Done